csharp/src/Apache.Arrow/Field.Builder.cs

```diff
@@ -70,6 +70,15 @@ public Builder Metadata(string key, string value)
                 return this;
             }
 
+            public Builder Metadata(IEnumerable<KeyValuePair<string, string>> dict)
```

Comment:
Should check for `null` dict.

Also `dict` should be spelled out. See [https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/general-naming-conventions](https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/general-naming-conventions)

> X DO NOT use abbreviations or contractions as part of identifier names.
> For example, use GetWindow rather than GetWin.